frontend/src/bundles/common/components/calendar/components/calendar-month.tsx

```diff
@@ -0,0 +1,38 @@
+import { useCallback } from 'react';
+
+import { type CalendarMonth } from '~/bundles/common/types/types';
+
+import styles from '../styles.module.scss';
+
+type Properties = {
+    month: CalendarMonth
+    onClick: (number_: CalendarMonth) => void;
+    selected: boolean;
+};
+
+const CalendarMonthComponent: React.FC<Properties> = ({
+    month,
+    onClick,
+    selected,
+}: Properties) => {
+    const handleClick = useCallback((): void => {
+        onClick(month);
+    }, [onClick, month]);
+
+    return (
+        <div
+            role="presentation"
+            className={
+                selected
+                    ? `${styles['date-picker__option']} ${styles['date-picker__selected']}`
+                    : styles['date-picker__option']
```

Comment:
Consider using `classnames` package
It's better to take the logic out of the layout
Also consider using the `date__picker__option` class name style to avoid adding square brackets